h5py/h5t.pyx

```diff
@@ -669,6 +669,12 @@ cdef class TypeOpaqueID(TypeID):
             free(buf)
 
     cdef object py_dtype(self):
+        cdef bytes tag = self.get_tag()
+        cdef bytes dt_str
+        if tag.startswith(b"NUMPY:"):
+            dt_str = tag[len(b"NUMPY:"):]
```

Comment:
Shouldn't `len(b"NUMPY:")` just be `6`, so Cython (and the C compiler) can optimize this?

Reply:
I didn't want to obfuscate it too much, but I can change it if needed. 😄

Reply:
You can add a comment on the line above if you feel it is unclear.